libmscore/measure.cpp

```diff
@@ -2676,13 +2682,24 @@ bool Measure::createEndBarLines()
                         ++aspan;
                         if (staff->show()) {          // update only if visible
                               bl->setSpan(aspan);
+                              // if the top staff was invisible
+                              // set spanFrom based on default for first visible staff
+                              if (show && spanFrom == unknownBarlineFrom)
```

Comment:
I think this to be the main change: is it the AND between the staff being shown but `spanFrom` being unset (`unkownBarlineFrom`) which tells us that the staff was previously hidden and it is shown now?

I wonder if it is not worth making `unkownBarlineFrom` a symbolic constant in `barline.h`, like `UNKNOWN_BARLINE_TO`.

For the rest, I do not see any obvious concern; a part from a few cosmetic improvements, the only other changes are some `spanFrom` and `spanTo` values made to depend on the number of staff lines, which surely improves the 1-line case and can hardly affect other types of staves.

Generally speaking this function, while not being particularly long, is rather convoluted and initially it took me a little while to understand it (this is why I added so many and detailed comments).